box-share-sdk/src/main/java/com/box/androidsdk/share/fragments/CollaborationsFragment.java

```diff
@@ -0,0 +1,270 @@
+        if (mCollaborations != null && mCollaborations.size() > 0) {
+            hideView(mNoCollaboratorsText);
+            showView(mCollaboratorsListView);
+            mCollaboratorsAdapter.setItems(mCollaborations);
+        } else {
+            hideView(mCollaboratorsListView);
+            showView(mNoCollaboratorsText);
+        }
+    }
+
+    public BoxCollaboration.Role[] getRoles() {
+        if (getFolder().getAllowedInviteeRoles() != null) {
+            return getFolder().getAllowedInviteeRoles().toArray(new BoxCollaboration.Role[getFolder().getAllowedInviteeRoles().size()]);
+        }
+        return null;
+    }
+
+    private BoxFutureTask.OnCompletedListener<BoxIteratorCollaborations> mCollaborationsListener =
+        new BoxFutureTask.OnCompletedListener<BoxIteratorCollaborations>() {
+            @Override
+            public void onCompleted(final BoxResponse<BoxIteratorCollaborations> response) {
+                dismissSpinner();
+                final Activity activity = getActivity();
+                if (activity == null) {
+                    return;
+                }
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (response.isSuccess() && getFolder() != null) {
+                            mCollaborations = response.getResult();
+                            updateUi();
+                        } else {
+                            BoxLogUtils.e(CollaborationsFragment.class.getName(), "Fetch Collaborators request failed",
+                                    response.getException());
+                            Toast.makeText(getActivity(), getString(R.string.box_sharesdk_network_error), Toast.LENGTH_LONG).show();
+                        }
+                    }
+                });
+            }
+        };
+
+    private BoxFutureTask.OnCompletedListener<BoxFolder> mRolesListener =
+        new BoxFutureTask.OnCompletedListener<BoxFolder>() {
+            @Override
+            public void onCompleted(final BoxResponse<BoxFolder> response) {
+                dismissSpinner();
+                final Activity activity = getActivity();
+                if (activity == null) {
+                    return;
+                }
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (response.isSuccess() && getFolder() != null) {
+                            BoxFolder folder = response.getResult();
+                            mShareItem = folder;
+                        } else {
+                            BoxLogUtils.e(CollaborationsFragment.class.getName(), "Fetch roles request failed",
+                                    response.getException());
+                            Toast.makeText(getActivity(), getString(R.string.box_sharesdk_network_error), Toast.LENGTH_LONG).show();
```

Comment:
Can you move these out into say the controller or something easy to override? Eventually we want to style errors toasts differently in the app.